reactor-core/src/main/java/reactor/core/publisher/FluxOnAssembly.java

```diff
@@ -441,6 +441,15 @@ public String getMessage() {
 					return super.getMessage();
 				}
 
+				if (Hooks.GLOBAL_TRACE) {
+					// attach dangling node, without a parent, to root
+					nodesPerId.forEach((id, node) -> {
+						if (node.parent == null && !node.children.isEmpty()) {
+							root.children.add(node);
+						}
+					});
+				}
+
```

Comment:
The above change was added, because the following "parallel" test cases were failing, but only with debug mode enabled. Debugging led me to this fix.

<img width="392" alt="image" src="https://github.com/user-attachments/assets/8559dee9-2ce1-4f4d-9bcf-f7a0ebc9417a">